lxd/storage/backend_lxd_patches.go

```diff
@@ -190,9 +191,29 @@ func patchBucketNames(b *lxdBackend) error {
 		return err
 	}
 
+	if len(buckets) == 0 {
+		return nil
+	}
+
+	reverter := revert.New()
+
+	// ListVolumes assumes that the pool is mounted for dir & btrfs drivers
+	_, err = b.Mount()
+	if err != nil {
+		return err
+	}
+
+	reverter.Add(func() {
+		_, err := b.Unmount()
+		if err != nil {
+			logger.Error("Failed to unmount pool", logger.Ctx{"pool": b.name, "err": err})
+		}
+	})
+
 	// Get list of volumes.
 	volumes, err := b.driver.ListVolumes()
 	if err != nil {
+		reverter.Fail()
```

Comment:
Should use `defer reverter.Fail()` rather than calling `reverter.Fail` on each error scenario.

Reply:
Duh, ty.